codec-http2/src/test/java/io/netty/handler/codec/http2/HttpToHttp2ConnectionHandlerTest.java

```diff
@@ -310,77 +250,55 @@ public void testHostIPv6FormRequestTargetHandled() throws Exception {
     }
 
     @Test
-    public void testHostNoSchemeFormRequestTargetHandled() throws Exception {
+    public void testHostFormRequestTargetHandled() throws Exception {
         bootstrapEnv(2, 1, 0);
         final FullHttpRequest request = new DefaultFullHttpRequest(HTTP_1_1, GET, "/");
         final HttpHeaders httpHeaders = request.headers();
         httpHeaders.setInt(HttpUtil.ExtensionHeaderNames.STREAM_ID.text(), 5);
-        // This is an "irregular" host in that the scheme is "localhost"
         httpHeaders.set(HttpHeaderNames.HOST, "localhost:80");
+        httpHeaders.set(HttpUtil.ExtensionHeaderNames.SCHEME.text(), "http");
         final Http2Headers http2Headers =
                 new DefaultHttp2Headers().method(new AsciiString("GET")).path(new AsciiString("/"))
-                .scheme(new AsciiString("localhost"));
-        ChannelPromise writePromise = newPromise();
-        ChannelFuture writeFuture = clientChannel.writeAndFlush(request, writePromise);
+                .scheme(new AsciiString("http")).authority(new AsciiString("localhost:80"));
 
-        assertTrue(writePromise.awaitUninterruptibly(WAIT_TIME_SECONDS, SECONDS));
-        assertTrue(writePromise.isSuccess());
-        assertTrue(writeFuture.awaitUninterruptibly(WAIT_TIME_SECONDS, SECONDS));
-        assertTrue(writeFuture.isSuccess());
-        awaitRequests();
-        verify(serverListener).onHeadersRead(any(ChannelHandlerContext.class), eq(5),
-                eq(http2Headers), eq(0), anyShort(), anyBoolean(), eq(0), eq(true));
-        verify(serverListener, never()).onDataRead(any(ChannelHandlerContext.class), anyInt(),
-                any(ByteBuf.class), anyInt(), anyBoolean());
+        ChannelPromise writePromise = newPromise();
+        verifyHeadersOnly(http2Headers, writePromise, clientChannel.writeAndFlush(request, writePromise));
     }
 
     @Test
-    public void testBadHostIPv4FormRequestTargetHandled() throws Exception {
-        // Invalid according to
-        // https://tools.ietf.org/html/rfc7230#section-2.7.1 -> https://tools.ietf.org/html/rfc3986#section-3
+    public void testHostIPv4FormRequestTargetHandled() throws Exception {
         bootstrapEnv(2, 1, 0);
         final FullHttpRequest request = new DefaultFullHttpRequest(HTTP_1_1, GET, "/");
         final HttpHeaders httpHeaders = request.headers();
         httpHeaders.setInt(HttpUtil.ExtensionHeaderNames.STREAM_ID.text(), 5);
         httpHeaders.set(HttpHeaderNames.HOST, "1.2.3.4:80");
-        ChannelPromise writePromise = newPromise();
-        ChannelFuture writeFuture = clientChannel.writeAndFlush(request, writePromise);
+        httpHeaders.set(HttpUtil.ExtensionHeaderNames.SCHEME.text(), "http");
+        final Http2Headers http2Headers =
+                new DefaultHttp2Headers().method(new AsciiString("GET")).path(new AsciiString("/"))
+                .scheme(new AsciiString("http")).authority(new AsciiString("1.2.3.4:80"));
```

Comment:
@trustin - I have a test for IPv6 here https://github.com/netty/netty/pull/4132/files#diff-f1929be678a866bc4d7a26ea2b0cd755R253. Is there a specific address or type you would like me to check?